docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md

```diff
@@ -0,0 +1,176 @@
+
+
+##  Overview
+
+Here is a high-level overview of the steps we will be taking in this tutorial:
+
+1. Setting up.
+2. Get current base fee from the network.  
+3. Set max priority fee.  
+4. Construct the transaction object.
+5. Sign and send the transaction to the RPC provider.
+6. Catch errors if any.
+Before we start writing and deploying our contract, we need to set up our environment. For that, we need to install the following:
+
+1. Node.js - Node.js is a JavaScript runtime environment that allows you to run JavaScript on the server-side. You can download it from https://nodejs.org/en/download/  
+
+2. npm - Node Package Manager is used to publish and install packages to and from the public npm registry or a private npm registry. Here is how to install it https://docs.npmjs.com/downloading-and-installing-node-js-and-npm. (Alternatively, you can use yarn instead of npm https://classic.yarnpkg.com/lang/en/docs/getting-started/)
+
+## Getting started
+
+Here we initialize Web3 and set an RPC url inside `const url;` , add our private key in the `const privateKey;` , then we specify the erc20 token inside the  
+ `const tokenAdress;`  
+   
+   
+There are two languages the demo is written in, typescript and javascript,
+typescript has `transferWithEIP1559(amountToTransfer: string): Promise<void> ` function and javascript has`async function transferWithEIP1559()` function.
+in typescript we have explicitly defined the type for the transaction object. Refer the tabs below.with specific type of parameters `amountToTransfer: string` and a defined return type `Promise<void>`.
+
+Inside the `async function transferWithEIP1559()`, we initialize a provider, then add account and get the current base fee from the network to store it inside of  
+`const gasPrice;` , next we set the maximum priority fee we are willing to allocate inside  `const maxPriorityFeePerGas;`, to do this we make use of utils package of web3.js, which is imported at the top.
+
+Construct the Transaction object:  
+To send the transaction details to the RPC provider for the transfer, we need to format the details in the tx object(transaction object). The tx object must contain the following:  
+from, to, value, gas, maxPriorityFeePerGas.
+Sign and send the transaction:  
+
+To sign a transaction we perform the `signTransaction` method on the account with the help of the web3.js lib.
+Next we use the `sendSignedTransaction` method with the `signedTx` parameters.
+
+Error handling:
+The catch block is activated if any exceptions or errors are thrown within the try block.
+this logs the error message to the console using `console.error('Error:', error).`
+
+
+<Tabs groupId='prog-lang' queryString>
+
+  <TabItem value='javascript' label='JavaScript'
+  	attributes={{className: 'javascript-tab'}}>
+
+``` 
+Javascript:
+
+const web3 = new Web3(new Web3.providers.WebsocketProvider(providerUrl));
+const Web3 = require('web3');
```

Comment:
web3 instance is created before its imported , pls add tested code. @sarthak1dev